### PR TITLE
CQM Converter updates for model changes

### PR DESCRIPTION
**Pull request by holmesie**
Includes changes for:

- Model fields now are camelCase
- Handling BSON::Document when converting
- Fixed case where expected_values are missing from extended_data
- Adjusted how dates are represented
- Made sure to remove 'infinity' end_date when converting back to HDS
- Minor documentation fix
- birthdate and expired patient characteristics now include codes

---

Pull requests into cqm-converter require the following. Submitter and reviewer should :white_check_mark: when done. For items that are not-applicable, note it's not-applicable ("N/A") and :white_check_mark:.

**Submitter:**
- [x] This pull request describes why these changes were made.
- [x] Internal ticket for this PR: https://jira.mitre.org/browse/TACO-56
- [x] Internal ticket links to this PR
- [x] Code diff has been done and been reviewed
- [x] Tests are included and test edge cases
- [x] Tests have been run locally and pass
- [x] If applicable, the library version number in `cqm-converter.gemspec` has been updated

**Reviewer 1:**

Name: @ljades 
- [x] Code is maintainable and reusable, reuses existing code and infrastructure where appropriate, and accomplishes the task’s purpose
- [x] The tests appropriately test the new code, including edge cases
- [x] You have tried to break the code

**Reviewer 2:**

Name: @okeefm 
- [x] Code is maintainable and reusable, reuses existing code and infrastructure where appropriate, and accomplishes the task’s purpose
- [x] The tests appropriately test the new code, including edge cases
- [x] You have tried to break the code

**Reviewer 3:**

Name: @jbradl11 or other Bonnie person
- [x] Code is maintainable and reusable, reuses existing code and infrastructure where appropriate, and accomplishes the task’s purpose
- [x] The tests appropriately test the new code, including edge cases
- [x] You have tried to break the code

### CODE REVIEW

File: lib/cqm/converter/qdm_patient.rb [hunk 1]
```diff
@@ -223,13 +226,13 @@ def unpack_components(hds_attrs)
 
     # Unpack diagnosis.
     def unpack_diagnosis(hds_attrs)
-      if hds_attrs.key?('diagnosis') && !hds_attrs['diagnosis'].nil?
+      if hds_attrs.key?('diagnosis') && !hds_attrs['diagnosis'].empty?
```
Comment (ljades) [inline lib/cqm/converter/qdm_patient.rb]: What was the issue being hit before this change, if I may ask?

File: lib/cqm/converter/qdm_patient.rb [hunk 2]
```diff
@@ -238,34 +241,34 @@ def unpack_diagnosis(hds_attrs)
       end
       # Remove diagnosis if principalDiagnosis is equivalent.
       return unless hds_attrs.key?('diagnosis') && hds_attrs.key?('principalDiagnosis')
-      return unless hds_attrs['diagnosis']['values'] && hds_attrs['diagnosis']['values'].first == hds_attrs['principalDiagnosis']
+      return unless hds_attrs['diagnosis']['values'] && Hash[hds_attrs['diagnosis']['values'].first.sort] == Hash[hds_attrs['principalDiagnosis'].sort]
```
Comment (ljades) [inline lib/cqm/converter/qdm_patient.rb]: This one too. Was the second condition never returning `true` the way prior?

File: lib/cqm/converter/utils.rb [hunk 3]
```diff
@@ -7,7 +7,8 @@ def self.qdm_codes_to_hds_codes(qdm_codes)
       codes = {}
       qdm_codes.each do |qdm_code|
         qdm_code = qdm_code.stringify_keys
-        code_system = qdm_code['code_system']
+        code_system = qdm_code['codeSystem']
+        code_system = qdm_code['code_system'] if qdm_code['code_system']
```
Comment (ljades) [inline lib/cqm/converter/utils.rb]: Are there instances where either of `codeSystem` or `code_system` can be set? I was under the impression that it was only one of the two.

File: lib/cqm/converter/qdm_patient.rb (hunk 1 above)
Comment (okeefm) [reply lib/cqm/converter/qdm_patient.rb]: I suspect what was happening here was that `hds_attrs['diagnosis']` was not `nil`, but didn't contain any information; either it was an empty `Hash` or an empty `Array`.

Comment (holmesie) [reply lib/cqm/converter/qdm_patient.rb]: Right

File: lib/cqm/converter/qdm_patient.rb (hunk 2 above)
Comment (holmesie) [reply lib/cqm/converter/qdm_patient.rb]: The way prior was for underscore variable names. This change makes sure the hash keys are sorted before comparison.

File: lib/cqm/converter/utils.rb (hunk 3 above)
Comment (holmesie) [reply lib/cqm/converter/utils.rb]: There are, this catches both cases.
